{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "ed8b134d-9810-46fd-9561-d307766d7563",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: bravado in c:\\users\\camila - bioinfo\\anaconda3\\lib\\site-packages (11.0.3)\n",
      "Requirement already satisfied: bravado-core>=5.16.1 in c:\\users\\camila - bioinfo\\anaconda3\\lib\\site-packages (from bravado) (6.1.1)\n",
      "Requirement already satisfied: msgpack in c:\\users\\camila - bioinfo\\anaconda3\\lib\\site-packages (from bravado) (1.0.3)\n",
      "Requirement already satisfied: python-dateutil in c:\\users\\camila - bioinfo\\anaconda3\\lib\\site-packages (from bravado) (2.8.2)\n",
      "Requirement already satisfied: pyyaml in c:\\users\\camila - bioinfo\\anaconda3\\lib\\site-packages (from bravado) (6.0)\n",
      "Requirement already satisfied: requests>=2.17 in c:\\users\\camila - bioinfo\\anaconda3\\lib\\site-packages (from bravado) (2.31.0)\n",
      "Requirement already satisfied: six in c:\\users\\camila - bioinfo\\anaconda3\\lib\\site-packages (from bravado) (1.16.0)\n",
      "Requirement already satisfied: simplejson in c:\\users\\camila - bioinfo\\anaconda3\\lib\\site-packages (from bravado) (3.19.2)\n",
      "Requirement already satisfied: monotonic in c:\\users\\camila - bioinfo\\anaconda3\\lib\\site-packages (from bravado) (1.6)\n",
      "Requirement already satisfied: typing-extensions in c:\\users\\camila - bioinfo\\anaconda3\\lib\\site-packages (from bravado) (4.7.1)\n",
      "Requirement already satisfied: jsonref in c:\\users\\camila - bioinfo\\anaconda3\\lib\\site-packages (from bravado-core>=5.16.1->bravado) (1.1.0)\n",
      "Requirement already satisfied: jsonschema[format-nongpl]>=2.5.1 in c:\\users\\camila - bioinfo\\anaconda3\\lib\\site-packages (from bravado-core>=5.16.1->bravado) (4.17.3)\n",
      "Requirement already satisfied: swagger-spec-validator>=2.0.1 in c:\\users\\camila - bioinfo\\anaconda3\\lib\\site-packages (from bravado-core>=5.16.1->bravado) (3.0.3)\n",
      "Requirement already satisfied: pytz in c:\\users\\camila - bioinfo\\anaconda3\\lib\\site-packages (from bravado-core>=5.16.1->bravado) (2022.7)\n",
      "Requirement already satisfied: charset-normalizer<4,>=2 in c:\\users\\camila - bioinfo\\anaconda3\\lib\\site-packages (from requests>=2.17->bravado) (2.0.4)\n",
      "Requirement already satisfied: idna<4,>=2.5 in c:\\users\\camila - bioinfo\\anaconda3\\lib\\site-packages (from requests>=2.17->bravado) (3.4)\n",
      "Requirement already satisfied: urllib3<3,>=1.21.1 in c:\\users\\camila - bioinfo\\anaconda3\\lib\\site-packages (from requests>=2.17->bravado) (1.26.16)\n",
      "Requirement already satisfied: certifi>=2017.4.17 in c:\\users\\camila - bioinfo\\anaconda3\\lib\\site-packages (from requests>=2.17->bravado) (2024.2.2)\n",
      "Requirement already satisfied: attrs>=17.4.0 in c:\\users\\camila - bioinfo\\anaconda3\\lib\\site-packages (from jsonschema[format-nongpl]>=2.5.1->bravado-core>=5.16.1->bravado) (22.1.0)\n",
      "Requirement already satisfied: pyrsistent!=0.17.0,!=0.17.1,!=0.17.2,>=0.14.0 in c:\\users\\camila - bioinfo\\anaconda3\\lib\\site-packages (from jsonschema[format-nongpl]>=2.5.1->bravado-core>=5.16.1->bravado) (0.18.0)\n",
      "Requirement already satisfied: fqdn in c:\\users\\camila - bioinfo\\anaconda3\\lib\\site-packages (from jsonschema[format-nongpl]>=2.5.1->bravado-core>=5.16.1->bravado) (1.5.1)\n",
      "Requirement already satisfied: isoduration in c:\\users\\camila - bioinfo\\anaconda3\\lib\\site-packages (from jsonschema[format-nongpl]>=2.5.1->bravado-core>=5.16.1->bravado) (20.11.0)\n",
      "Requirement already satisfied: jsonpointer>1.13 in c:\\users\\camila - bioinfo\\anaconda3\\lib\\site-packages (from jsonschema[format-nongpl]>=2.5.1->bravado-core>=5.16.1->bravado) (2.1)\n",
      "Requirement already satisfied: rfc3339-validator in c:\\users\\camila - bioinfo\\anaconda3\\lib\\site-packages (from jsonschema[format-nongpl]>=2.5.1->bravado-core>=5.16.1->bravado) (0.1.4)\n",
      "Requirement already satisfied: rfc3986-validator>0.1.0 in c:\\users\\camila - bioinfo\\anaconda3\\lib\\site-packages (from jsonschema[format-nongpl]>=2.5.1->bravado-core>=5.16.1->bravado) (0.1.1)\n",
      "Requirement already satisfied: uri-template in c:\\users\\camila - bioinfo\\anaconda3\\lib\\site-packages (from jsonschema[format-nongpl]>=2.5.1->bravado-core>=5.16.1->bravado) (1.3.0)\n",
      "Requirement already satisfied: webcolors>=1.11 in c:\\users\\camila - bioinfo\\anaconda3\\lib\\site-packages (from jsonschema[format-nongpl]>=2.5.1->bravado-core>=5.16.1->bravado) (1.13)\n",
      "Requirement already satisfied: arrow>=0.15.0 in c:\\users\\camila - bioinfo\\anaconda3\\lib\\site-packages (from isoduration->jsonschema[format-nongpl]>=2.5.1->bravado-core>=5.16.1->bravado) (1.2.3)\n"
     ]
    }
   ],
   "source": [
    "!pip install bravado"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "90128532-f1a0-4db0-8178-d5d71595e65b",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "from bravado.client import SwaggerClient"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "a0b02d0d-93b1-4142-95a8-8cd6a28cb9af",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "cbioportal = SwaggerClient.from_url('https://www.cbioportal.org/api/v2/api-docs',\n",
    "                                    config={\"validate_requests\":False,\"validate_responses\":False,\"validate_swagger_spec\": False})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "e31f134c-a675-4367-950c-e6d32c222a85",
   "metadata": {},
   "outputs": [],
   "source": [
    "def mutation_to_maf(path):\n",
    "\n",
    "    combined = pd.read_csv(path, sep=\"\\t\", dtype='object')\n",
    "    studies = list(combined['Study ID'].unique())\n",
    "    \n",
    "    mutations_list = []\n",
    "    df_list = []\n",
    "    \n",
    "    #carregando as mutações e os sample ID a partir de uma lista de estudos\n",
    "    for studie in studies:\n",
    "        mutations = cbioportal.Mutations.getMutationsInMolecularProfileBySampleListIdUsingGET(\n",
    "            molecularProfileId='{}_mutations'.format(studie),\n",
    "            sampleListId='{}_all'.format(studie),\n",
    "            projection='DETAILED').result()\n",
    "        \n",
    "        mutations_list.append(mutations) \n",
    "\n",
    "    #transformando as mutações em dataframes\n",
    "    for mutations in mutations_list:\n",
    "        df = pd.DataFrame.from_dict([dict(\n",
    "            {k:getattr(m,k) for k in dir(m)},\n",
    "            **{k:getattr(m.gene,k) for k in dir(m.gene)}) for m in mutations])\n",
    "        df.dropna()\n",
    "        df_list.append(df)\n",
    "    \n",
    "    studie_mutations = pd.concat(df_list, axis=0) #concatenando os dataframes\n",
    "    \n",
    "    #escolhendo somente as colunas exigidas pelo maftools\n",
    "    maf_ = studie_mutations[[#'alleleSpecificCopyNumber', 'aminoAcidChange', 'center', \n",
    "    'chr',\n",
    "       #'driverFilter', 'driverFilterAnnotation', 'driverTiersFilter',\n",
    "       #'driverTiersFilterAnnotation', \n",
    "    'endPosition', #'entrezGeneId', 'gene',\n",
    "       #'keyword', 'molecularProfileId', 'mutationStatus', \n",
    "    'mutationType',\n",
    "       #'namespaceColumns', 'ncbiBuild', 'normalAltCount', 'normalRefCount',\n",
    "       #'patientId', \n",
    "    'proteinChange', #'proteinPosEnd', 'proteinPosStart',\n",
    "       'referenceAllele',# 'refseqMrnaId', \n",
    "    'sampleId', 'startPosition',\n",
    "       'studyId', 'tumorAltCount', 'tumorRefCount', \n",
    "    #'uniquePatientKey',\n",
    "       #'uniqueSampleKey', 'validationStatus', \n",
    "    'variantAllele', \n",
    "    'variantType',\n",
    "       #'geneticEntityId', 'type'\n",
    "    'hugoGeneSymbol']]\n",
    "    \n",
    "    maf = maf_.drop_duplicates() \n",
    "    \n",
    "\n",
    "    #renomeando para o maftools\n",
    "    maf = maf.rename(columns={'hugoGeneSymbol':'Hugo_Symbol',\n",
    "                    'sampleId':'Tumor_Sample_Barcode',\n",
    "                    'proteinChange': 'Protein_Change',\n",
    "                    'mutationType': 'Variant_Classification',\n",
    "                    'variantType': 'Variant_Type',\n",
    "                    'startPosition': 'Start_Position',\n",
    "                    'endPosition': 'End_Position',\n",
    "                    'referenceAllele':'Reference_Allele',\n",
    "                    'variantAllele': 'Tumor_Seq_Allele2',\n",
    "                          'studyId':'Study_ID'})\n",
    "    \n",
    "    return maf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "e3844afa",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "scrolled": true,
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GENE</th>\n",
       "      <th>CNV</th>\n",
       "      <th>DEL</th>\n",
       "      <th>DUP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ASXL1</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BRAF</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CALR</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CBL</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CDKN2A</td>\n",
       "      <td>1</td>\n",
       "      <td>66</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>CDKN2B</td>\n",
       "      <td>1</td>\n",
       "      <td>57</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>ETV6</td>\n",
       "      <td>0</td>\n",
       "      <td>29</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>EZH2</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>FBXW7</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>GATA1</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>34</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>IKZF1</td>\n",
       "      <td>0</td>\n",
       "      <td>38</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>IL7R</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>JAK2</td>\n",
       "      <td>1</td>\n",
       "      <td>19</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>KIT</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>KRAS</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>NF1</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>NPM1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>PHF6</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>PTEN</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>52</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>SETBP1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>SH2B3</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>SRSF2</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>TET2</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>TP53</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>U2AF1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>47</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>ZRSR2</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>36</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>ABL1</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>NOTCH1</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>CREBBP</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>IDH2</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>WT1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>CEBPA</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>CHR_EVENT</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>CSF3R</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>FLT3</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>IDH1</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>JAK1</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>MPL</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>NRAS</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>SF3B1</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>DNMT3A</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>HRAS</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>PTPN11</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         GENE  CNV  DEL  DUP\n",
       "0       ASXL1    0    5    3\n",
       "1        BRAF    1    6    5\n",
       "2        CALR    0    3    1\n",
       "3         CBL    0    6    5\n",
       "4      CDKN2A    1   66   13\n",
       "5      CDKN2B    1   57   10\n",
       "6        ETV6    0   29    3\n",
       "7        EZH2    0    7    5\n",
       "8       FBXW7    0    4   23\n",
       "9       GATA1    0    5   34\n",
       "10      IKZF1    0   38    1\n",
       "11       IL7R    1    1    7\n",
       "12       JAK2    1   19   13\n",
       "13        KIT    0    1   22\n",
       "14       KRAS    0   10    1\n",
       "15        NF1    0    6   22\n",
       "16       NPM1    1    0    9\n",
       "17       PHF6    0    7   31\n",
       "18       PTEN    0    2   29\n",
       "19      RUNX1    0    1   52\n",
       "20     SETBP1    0    0   18\n",
       "21      SH2B3    1    4    3\n",
       "22      SRSF2    0    1   22\n",
       "23       TET2    0    3   22\n",
       "24       TP53    0    1   20\n",
       "25      U2AF1    0    1   47\n",
       "26      ZRSR2    0    6   36\n",
       "27       ABL1    1    4    8\n",
       "28     NOTCH1    0    4    9\n",
       "29     CREBBP    0    1    2\n",
       "30       IDH2    0    3    0\n",
       "31        WT1    0    0    5\n",
       "32      CEBPA    0    3    0\n",
       "33  CHR_EVENT    0    1    0\n",
       "34      CSF3R    1    3    1\n",
       "35       FLT3    0    3    1\n",
       "36       IDH1    0    3    1\n",
       "37       JAK1    1    3    1\n",
       "38        MPL    1    3    1\n",
       "39       NRAS    0    3    1\n",
       "40      SF3B1    0    3    1\n",
       "41     DNMT3A    1    0    2\n",
       "42       HRAS    0    0    3\n",
       "43     PTPN11    0    0    2"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "list_genes = pd.read_csv(\"genes_leukemia/BALL.csv\")\n",
    "list_genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "4a27f8c2-1e4d-4e36-b695-7fdbde857cd4",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\Camila - bioinfo\\AppData\\Local\\Temp\\ipykernel_3156\\1916385685.py:26: FutureWarning: The behavior of DataFrame concatenation with empty or all-NA entries is deprecated. In a future version, this will no longer exclude empty or all-NA columns when determining the result dtypes. To retain the old behavior, exclude the relevant entries before the concat operation.\n",
      "  studie_mutations = pd.concat(df_list, axis=0) #concatenando os dataframes\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>End_Position</th>\n",
       "      <th>Variant_Classification</th>\n",
       "      <th>Protein_Change</th>\n",
       "      <th>Reference_Allele</th>\n",
       "      <th>Tumor_Sample_Barcode</th>\n",
       "      <th>Start_Position</th>\n",
       "      <th>Study_ID</th>\n",
       "      <th>tumorAltCount</th>\n",
       "      <th>tumorRefCount</th>\n",
       "      <th>Tumor_Seq_Allele2</th>\n",
       "      <th>Variant_Type</th>\n",
       "      <th>Hugo_Symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>4</td>\n",
       "      <td>106156043</td>\n",
       "      <td>Frame_Shift_Del</td>\n",
       "      <td>Q317Rfs*30</td>\n",
       "      <td>C</td>\n",
       "      <td>aml_ohsu_2018_09-00705</td>\n",
       "      <td>106156043</td>\n",
       "      <td>aml_ohsu_2018</td>\n",
       "      <td>68.0</td>\n",
       "      <td>151.0</td>\n",
       "      <td>-</td>\n",
       "      <td>DEL</td>\n",
       "      <td>TET2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>4</td>\n",
       "      <td>106190830</td>\n",
       "      <td>Frame_Shift_Del</td>\n",
       "      <td>V1371Sfs*77</td>\n",
       "      <td>G</td>\n",
       "      <td>aml_ohsu_2018_09-00705</td>\n",
       "      <td>106190830</td>\n",
       "      <td>aml_ohsu_2018</td>\n",
       "      <td>28.0</td>\n",
       "      <td>74.0</td>\n",
       "      <td>-</td>\n",
       "      <td>DEL</td>\n",
       "      <td>TET2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>5</td>\n",
       "      <td>170837544</td>\n",
       "      <td>Frame_Shift_Ins</td>\n",
       "      <td>W288Cfs*12</td>\n",
       "      <td>-</td>\n",
       "      <td>aml_ohsu_2018_09-00705</td>\n",
       "      <td>170837543</td>\n",
       "      <td>aml_ohsu_2018</td>\n",
       "      <td>17.0</td>\n",
       "      <td>59.0</td>\n",
       "      <td>TCTG</td>\n",
       "      <td>INS</td>\n",
       "      <td>NPM1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>2</td>\n",
       "      <td>25457242</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>R882H</td>\n",
       "      <td>C</td>\n",
       "      <td>aml_ohsu_2018_10-00136</td>\n",
       "      <td>25457242</td>\n",
       "      <td>aml_ohsu_2018</td>\n",
       "      <td>11.0</td>\n",
       "      <td>28.0</td>\n",
       "      <td>T</td>\n",
       "      <td>SNP</td>\n",
       "      <td>DNMT3A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>11</td>\n",
       "      <td>32456652</td>\n",
       "      <td>Frame_Shift_Del</td>\n",
       "      <td>A82Pfs*15</td>\n",
       "      <td>C</td>\n",
       "      <td>aml_ohsu_2018_10-00136</td>\n",
       "      <td>32456652</td>\n",
       "      <td>aml_ohsu_2018</td>\n",
       "      <td>18.0</td>\n",
       "      <td>51.0</td>\n",
       "      <td>-</td>\n",
       "      <td>DEL</td>\n",
       "      <td>WT1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1835</th>\n",
       "      <td>11</td>\n",
       "      <td>1272152</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>T4681I</td>\n",
       "      <td>C</td>\n",
       "      <td>TCGA-AB-2988-03</td>\n",
       "      <td>1272152</td>\n",
       "      <td>laml_tcga_pub</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>T</td>\n",
       "      <td>SNP</td>\n",
       "      <td>MUC5B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1836</th>\n",
       "      <td>7</td>\n",
       "      <td>75051473</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>G688R</td>\n",
       "      <td>C</td>\n",
       "      <td>TCGA-AB-2803-03</td>\n",
       "      <td>75051473</td>\n",
       "      <td>laml_tcga_pub</td>\n",
       "      <td>1.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>G</td>\n",
       "      <td>SNP</td>\n",
       "      <td>POM121C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1837</th>\n",
       "      <td>16</td>\n",
       "      <td>33629937</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>H54Q</td>\n",
       "      <td>C</td>\n",
       "      <td>TCGA-AB-2980-03</td>\n",
       "      <td>33629937</td>\n",
       "      <td>laml_tcga_pub</td>\n",
       "      <td>110.0</td>\n",
       "      <td>557.0</td>\n",
       "      <td>G</td>\n",
       "      <td>SNP</td>\n",
       "      <td>IGHV3OR16-13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1838</th>\n",
       "      <td>7</td>\n",
       "      <td>148964299</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>R220K</td>\n",
       "      <td>G</td>\n",
       "      <td>TCGA-AB-2868-03</td>\n",
       "      <td>148964299</td>\n",
       "      <td>laml_tcga_pub</td>\n",
       "      <td>103.0</td>\n",
       "      <td>248.0</td>\n",
       "      <td>A</td>\n",
       "      <td>SNP</td>\n",
       "      <td>ZNF783</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1839</th>\n",
       "      <td>9</td>\n",
       "      <td>100085148</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>R442H</td>\n",
       "      <td>G</td>\n",
       "      <td>TCGA-AB-2917-03</td>\n",
       "      <td>100085148</td>\n",
       "      <td>laml_tcga_pub</td>\n",
       "      <td>30.0</td>\n",
       "      <td>128.0</td>\n",
       "      <td>A</td>\n",
       "      <td>SNP</td>\n",
       "      <td>CCDC180</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>47868 rows × 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     chr  End_Position Variant_Classification Protein_Change Reference_Allele  \\\n",
       "0      4     106156043        Frame_Shift_Del     Q317Rfs*30                C   \n",
       "1      4     106190830        Frame_Shift_Del    V1371Sfs*77                G   \n",
       "2      5     170837544        Frame_Shift_Ins     W288Cfs*12                -   \n",
       "3      2      25457242      Missense_Mutation          R882H                C   \n",
       "4     11      32456652        Frame_Shift_Del      A82Pfs*15                C   \n",
       "...   ..           ...                    ...            ...              ...   \n",
       "1835  11       1272152      Missense_Mutation         T4681I                C   \n",
       "1836   7      75051473      Missense_Mutation          G688R                C   \n",
       "1837  16      33629937      Missense_Mutation           H54Q                C   \n",
       "1838   7     148964299      Missense_Mutation          R220K                G   \n",
       "1839   9     100085148      Missense_Mutation          R442H                G   \n",
       "\n",
       "        Tumor_Sample_Barcode  Start_Position       Study_ID  tumorAltCount  \\\n",
       "0     aml_ohsu_2018_09-00705       106156043  aml_ohsu_2018           68.0   \n",
       "1     aml_ohsu_2018_09-00705       106190830  aml_ohsu_2018           28.0   \n",
       "2     aml_ohsu_2018_09-00705       170837543  aml_ohsu_2018           17.0   \n",
       "3     aml_ohsu_2018_10-00136        25457242  aml_ohsu_2018           11.0   \n",
       "4     aml_ohsu_2018_10-00136        32456652  aml_ohsu_2018           18.0   \n",
       "...                      ...             ...            ...            ...   \n",
       "1835         TCGA-AB-2988-03         1272152  laml_tcga_pub            1.0   \n",
       "1836         TCGA-AB-2803-03        75051473  laml_tcga_pub            1.0   \n",
       "1837         TCGA-AB-2980-03        33629937  laml_tcga_pub          110.0   \n",
       "1838         TCGA-AB-2868-03       148964299  laml_tcga_pub          103.0   \n",
       "1839         TCGA-AB-2917-03       100085148  laml_tcga_pub           30.0   \n",
       "\n",
       "      tumorRefCount Tumor_Seq_Allele2 Variant_Type   Hugo_Symbol  \n",
       "0             151.0                 -          DEL          TET2  \n",
       "1              74.0                 -          DEL          TET2  \n",
       "2              59.0              TCTG          INS          NPM1  \n",
       "3              28.0                 T          SNP        DNMT3A  \n",
       "4              51.0                 -          DEL           WT1  \n",
       "...             ...               ...          ...           ...  \n",
       "1835            0.0                 T          SNP         MUC5B  \n",
       "1836            9.0                 G          SNP       POM121C  \n",
       "1837          557.0                 G          SNP  IGHV3OR16-13  \n",
       "1838          248.0                 A          SNP        ZNF783  \n",
       "1839          128.0                 A          SNP       CCDC180  \n",
       "\n",
       "[47868 rows x 13 columns]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "leukemia = mutation_to_maf(\"combined_leukemia_studies.tsv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "c340284f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['NPM1', 'WT1', 'SRSF2', 'IDH1', 'PHF6', 'JAK2', 'ZRSR2', 'CDKN2A',\n",
       "       'PTEN', 'GATA1'], dtype=object)"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "subset_leuk = leukemia[leukemia['Hugo_Symbol'].isin(list(list_genes['GENE']))]\n",
    "subset_leuk['Hugo_Symbol'].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "9b03a728-4874-4e63-9ea7-6426eeb95901",
   "metadata": {},
   "outputs": [],
   "source": [
    "subset_leuk.to_csv(\"TALL_maf.txt\", sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c2f4d612",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
